elastic/kibana
# [Profiling,Infra,APM] Disable Profiling integration by default

Closes https://github.com/elastic/kibana/issues/175016

## Summary

This PR disables the Profiling integration in Infra and APM by default on the plugin configuration level because this integration require users to first configure the main `profiling` plugin. On-prem users will have to manually enable both integrations once they enabled the Universal Profiling for their hosts. Cloud users will have Infra and APM integrations enabled by default because on Cloud instances Universal Profiling is already configured. A PR for the default Cloud settings will follow after this one is merged.

Changes I've made:
* Disabled the Infra integration be default
* Introduced a new APM feature flag for the Profiling integration
* Made sure all the places in APM that rely on Profiling integration respect the new feature flag
* Fixed a bug in APM when Universal Profiling was shown even though the integration was disabled in UI settings

https://github.com/elastic/kibana/assets/793851/65dfbb5b-1850-4d18-a92a-6ad59e0436a3

## How To Test

1. Checkout locally
2. Make sure you don't have `xpack.infra.featureFlags.profilingEnabled` already enabled in `kibana.yml`
3. Open kibana and make sure you don't see "Universal Profiling" tab in Host and Service details
4. Enabled both flags in `kibana.yml`: `xpack.infra.featureFlags.profilingEnabled` and `xpack.apm.featureFlags.profilingIntegrationAvailable: true`
5. Check that now you see "Universal Profiling" tab in the details screens in both Infra and APM
6. Go to Infra settings view and disable the Profiling integration, make sure the tab disappears
7. 6. Go to APM settings view and disable the Profiling integration, make sure the tab disappears
